### PR TITLE
Fabo/2044 uatoms

### LINKED ISSUE #2044: Big numbers means big responsibilities
**Describe the bug**
Big numbers are not displayed nicely on the header with a 15" screen

**To Reproduce**
1. Login when you are rich AF
2. Look at your money and dislike the way they are printed, they make you look poorer

**Expected behavior**
See 19M or some other cool expandable shorthand that on hover or on click show the whole number

**Screenshots**
screenshot 2019-02-21 at 11 32 43

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Chrome
 - Version 72

We have to decide if the fix I suggest make sense for everyone and/or what else we can do to show massive numbers, the chain will grow so also block numbers will hit big numbers

### PR DESCRIPTION
**Pull request by faboweb**
Closes #2044 

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

image
image

### CODE REVIEW

**Comment by fedekunze** [comment]
Let's also use `short()` from `scripts/num.js`  for the balance header to display `K`, `M`, `B`

**Comment by faboweb** [comment]
> we should allow for decimals on the Action Modal if we're displaying atoms there

nice catch :)

File: app/src/renderer/components/common/TmField.vue [hunk 1]
```diff
@@ -159,7 +159,7 @@ export default {
       let formattedValue = value
 
       if (this.type === `number`) {
-        formattedValue = Number(value)
+        formattedValue = value.trim()
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmField.vue]: this seems odd, trim doesn't work on numbers

Comment (faboweb) [reply app/src/renderer/components/common/TmField.vue]: value is always a string as it comes from `$event.target.value`.

**Comment by fedekunze** [comment]
tests failing

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,13 +9,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
 
+- [\#2044](https://github.com/cosmos/voyager/issues/2044) convert uatoms to atoms @faboweb
 - Refactored tests: LiProposal @jbibla
 
 ## [1.0.0-beta.9] - 2019-02-28
 
 ### Fixed
 
 - querying delegations was caught in a loop @faboweb
+- resetting forms when action modal closes @faboweb
```
Comment (jbibla) [inline CHANGELOG.md]: ❤️

File: app/src/renderer/components/governance/ModalVote.vue [hunk 3]
```diff
@@ -7,6 +7,7 @@
     title="Vote"
     class="modal-vote"
     submission-error-prefix="Voting failed"
+    @close="$v.$reset()"
```
Comment (jbibla) [inline app/src/renderer/components/governance/ModalVote.vue]: do you need to add this to each modal if it's already on the action modal parent?

File: app/src/renderer/components/staking/UndelegationModal.vue [hunk 4]
```diff
@@ -131,7 +133,7 @@ export default {
     },
     async submitForm(submitType, password) {
       await this.$store.dispatch(`submitUnbondingDelegation`, {
-        amount: -this.amount,
+        amount: -uatoms(this.amount),
```
Comment (jbibla) [inline app/src/renderer/components/staking/UndelegationModal.vue]: it's very weird to me that the REST API is expecting a negative number

Comment (jbibla) [reply app/src/renderer/components/staking/UndelegationModal.vue]: undelegating a negative number is a double negative 😖

Comment (fedekunze) [reply app/src/renderer/components/staking/UndelegationModal.vue]: this is not from the API, just from our delegation or send module

**Pull request merged**